src/mono/wasm/runtime/logging.ts

```diff
@@ -60,10 +61,11 @@ export function mono_wasm_symbolicate_string(message: string): string {
     }
 }
 
-export function mono_wasm_stringify_as_error_with_stack(err: Error | string): string {
+export function mono_wasm_stringify_as_error_with_stack(err: Error | ManagedError | string): string {
     let errObj: any = err;
-    if (!(err instanceof Error))
-        errObj = new Error(err);
+    if (!(errObj instanceof Error) && !(errObj instanceof ManagedError)) {
```

Comment:
Is managederror not derived from error?

Reply:
I have not realized that `instanceof` follows prototype chain, thanks!